src/brod_client.erl

```diff
@@ -418,21 +418,39 @@ shutdown_pid(_) ->
 
 -spec get_partition_worker(client(), partition_worker_key()) ->
         {ok, pid()} | {error, get_worker_error()}.
-get_partition_worker(ClientPid, Key) when is_pid(ClientPid) ->
-  case erlang:process_info(ClientPid, registered_name) of
-    {registered_name, ClientId} when is_atom(ClientId) ->
-      get_partition_worker(ClientId, Key);
+get_partition_worker(Client, Key) when is_atom(Client) ->
+  case erlang:whereis(Client) of
+    ClientPid when is_pid(ClientPid) ->
```

Comment:
Not sure if you have a preference here. We could call `get_partition_worker/2` and add a function override match on `undefined` or `_` now that we have a pid or know that it's a bad pid. If we have a valid pid, then we would then immediately be looking up its registered name. 😆